File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/go/service.go
package service

import (
	"myapp/database"
	"myapp/models"
)

func CreateDog(name string) *models.Dog {
	dog := models.NewDog(name)
	if !dog.Validate() {
		panic("invalid dog")
	}
	conn := database.GetConnection()
	conn.Execute("INSERT INTO dogs VALUES (?)")
	conn.Commit()
	return dog
}

func CreateCat(name string) *models.Cat {
	cat := models.NewCat(name)
	if !cat.Validate() {
		panic("invalid cat")
	}
	conn := database.GetConnection()
	conn.Execute("INSERT INTO cats VALUES (?)")
	conn.Commit()
	return cat
}

func TransferAnimal(name string, shelter *models.Shelter) {
	txn := database.NewTransaction(database.GetConnection())
	txn.Execute("UPDATE animals SET shelter_id = ?")
	shelter.Add(name)
	txn.Commit()
}

func ListAnimals() []string {
	conn := database.GetConnection()
	return conn.Execute("SELECT * FROM animals")
}

File: dot_local/share/codex/skills/sem/references/sem/crates/sem-core/tests/fixtures/scope_test/rust/service.rs
use crate::models::{Dog, Cat, Shelter};
use crate::database::{get_connection, Transaction};

pub fn create_dog(name: String) -> Dog {
    let dog = Dog::new(name);
    if !dog.validate() {
        panic!("invalid dog");
    }
    let conn = get_connection();
    conn.execute("INSERT INTO dogs VALUES (?)");
    conn.commit();
    dog
}

pub fn create_cat(name: String) -> Cat {
    let cat = Cat::new(name);
    if !cat.validate() {
        panic!("invalid cat");
    }
    let conn = get_connection();
    conn.execute("INSERT INTO cats VALUES (?)");
    conn.commit();
    cat
}

pub fn transfer_animal(name: String, shelter: &mut Shelter) {
    let txn = Transaction::new(get_connection());
    txn.execute("UPDATE animals SET shelter_id = ?");
    shelter.add(name);
    txn.commit();
}

pub fn list_animals() -> Vec<String> {
    let conn = get_connection();
    conn.execute("SELECT * FROM animals")
}
